traits/ctraits.c

```diff
@@ -2190,76 +2190,55 @@ call_notifiers(
     // Do nothing if the user has explicitly requested no traits notifications
     // to be sent.
     if ((obj->flags & HASTRAITS_NO_NOTIFY)) {
-        goto exit2;
+        goto exit;
```

Comment:
This is probably a matter of personal taste, but I'd find this clearer if it were simply a `Py_DECREF(args)` followed by `return 0` (and if the other `goto` were similarly replaced); with this refactor, I don't think the `goto`s save much in the way of code duplication.